configure

```diff
@@ -431,14 +431,29 @@ def is_arch_armv7():
   return ('__ARM_ARCH_7__' in cc_macros_cache or
           '__ARM_ARCH_7A__' in cc_macros_cache or
           '__ARM_ARCH_7R__' in cc_macros_cache or
-          '__ARM_ARCH_7M__' in cc_macros_cache)
+          '__ARM_ARCH_7M__' in cc_macros_cache or
+          '__ARM_ARCH_7S__' in cc_macros_cache)
 
 
 def is_arch_armv6():
   """Check for ARMv6 instructions"""
   cc_macros_cache = cc_macros()
   return ('__ARM_ARCH_6__' in cc_macros_cache or
-          '__ARM_ARCH_6M__' in cc_macros_cache)
+          '__ARM_ARCH_6M__' in cc_macros_cache or
+          '__ARM_ARCH_6J__' in cc_macros_cache or
+          '__ARM_ARCH_6K__' in cc_macros_cache or
+          '__ARM_ARCH_6Z__' in cc_macros_cache or
+          '__ARM_ARCH_6ZK__' in cc_macros_cache or
+          '__ARM_ARCH_6T2__' in cc_macros_cache)
+
+def is_arch_armv5():
+  """Check for ARMv5 instructions"""
+  cc_macros_cache = cc_macros()
+  return ('__ARM_ARCH_5__' in cc_macros_cache or
+  	      '__ARM_ARCH_5E__' in cc_macros_cache or
+  	      '__ARM_ARCH_5T__' in cc_macros_cache or
+  	      '__ARM_ARCH_5TE__' in cc_macros_cache or
+  	      '__ARM_ARCH_5TEJ__' in cc_macros_cache)
```

Comment:
I add them as a plus. I know is not used but is a nice to have. If you think is not necessary I have not problem with remove them.